src/app/api/sync-bot-source/route.ts

```diff
@@ -343,10 +349,114 @@ async function saveEmbeddings(
   content: string,
   embeddings: number[],
 ) {
+  if (!content) {
+    return
+  }
+
   await client.insert(schema.botSourceExtractedDataVector).values({
     id: uuidv7(),
     botSourceExtractedDataId: bsDataId,
     content,
     vector: embeddings,
   })
 }
+
+async function syncBotSourceFile(
+  bs: InferSelectModel<typeof schema.botSources>,
+) {
+  if (bs.statusId !== BotSourceStatusEnum.Created) {
+    throw new Error('Invalid bot source status')
+  }
+
+  if (!bs.url) {
+    throw new Error('URL is required')
+  }
+
+  const bsId = bs.id
+
+  try {
+    await db
+      .update(schema.botSources)
+      .set({ statusId: BotSourceStatusEnum.Crawling })
+      .where(eq(schema.botSources.id, bs.id))
+
+    const fileURL = bs.url
+
+    // Fetch file content
+    const res = await fetch(fileURL)
+    const sourcesText = await res.blob()
+
+    const loader = await formatFileLoader(sourcesText)
+
+    if (!loader) {
+      return
+    }
+
+    const docs = await loader.load()
+    const splitter = new RecursiveCharacterTextSplitter({
```

Comment:
Seem we're splitting manually ? 
Have you checked this? Is the spitter working as expected? 
Could it create the huge paragraph if the content have no splitter?